File: benchmarks/compare/raptor/go/rangequery_test.go
package raptor

import "testing"

func TestRangeQuery(t *testing.T) {
	journeyFactory := NewJourneyFactory()

	t.Run("performs profile queries", func(t *testing.T) {
		trips := []*Trip{
			tripOf(st("A", nil, p(1000)), st("B", p(1030), p(1035)), st("C", p(1100), nil)),
			tripOf(st("A", nil, p(1100)), st("B", p(1130), p(1135)), st("C", p(1200), nil)),
			tripOf(st("A", nil, p(1200)), st("B", p(1230), p(1235)), st("C", p(1300), nil)),
		}
		raptor := CreateRaptorAlgorithm(trips, nil, nil, nil)
		q := NewRangeQuery(raptor, journeyFactory, 0, nil)
		result := q.Plan("A", "C", NewSearchDate("2018-10-16"), 0, 0)
		setDefaultTrip(result)

		expected := []Journey{
			j([]StopTime{st("A", nil, p(1000)), st("B", p(1030), p(1035)), st("C", p(1100), nil)}),
			j([]StopTime{st("A", nil, p(1100)), st("B", p(1130), p(1135)), st("C", p(1200), nil)}),
			j([]StopTime{st("A", nil, p(1200)), st("B", p(1230), p(1235)), st("C", p(1300), nil)}),
		}
		assertJourneys(t, result, expected)
	})

	t.Run("does not share bestArrivals or routeScanner", func(t *testing.T) {
		trips := []*Trip{
			tripOf(st("A", nil, p(1359)), st("C", p(1501), nil)),
			tripOf(st("A", nil, p(1400)), st("B", p(1430), nil)),
			tripOf(st("B", nil, p(1430)), st("C", p(1500), nil)),
		}
		raptor := CreateRaptorAlgorithm(trips, nil, nil, nil)
		q := NewRangeQuery(raptor, journeyFactory, 0, nil)
		result := q.Plan("A", "C", NewSearchDate("2018-10-16"), 0, 0)
		setDefaultTrip(result)

		ac := func() Journey {
			return j([]StopTime{st("A", nil, p(1400)), st("B", p(1430), nil)},
				[]StopTime{st("B", nil, p(1430)), st("C", p(1500), nil)})
		}
		expected := []Journey{
			j([]StopTime{st("A", nil, p(1359)), st("C", p(1501), nil)}),
			ac(),
			ac(),
		}
		assertJourneys(t, result, expected)
	})
}

File: benchmarks/compare/raptor/node/test/RangeQuery.test.js
import { describe, it } from "node:test";
import { expect } from "./expect.js";
import { j, setDefaultTrip, st, t } from "./util.js";
import { JourneyFactory } from "../src/results/JourneyFactory.js";
import { RaptorAlgorithmFactory } from "../src/raptor/RaptorAlgorithmFactory.js";
import { RangeQuery } from "../src/query/RangeQuery.js";

describe("RangeQuery", () => {
  const journeyFactory = new JourneyFactory();

  it("performs profile queries", () => {
    const trips = [
      t(st("A", null, 1000), st("B", 1030, 1035), st("C", 1100, null)),
      t(st("A", null, 1100), st("B", 1130, 1135), st("C", 1200, null)),
      t(st("A", null, 1200), st("B", 1230, 1235), st("C", 1300, null))
    ];

    const raptor = RaptorAlgorithmFactory.create(trips, {}, {});
    const query = new RangeQuery(raptor, journeyFactory);
    const result = query.plan("A", "C", new Date("2018-10-16"));

    setDefaultTrip(result);

    expect(result).toEqual([
      j([st("A", null, 1000), st("B", 1030, 1035), st("C", 1100, null)]),
      j([st("A", null, 1100), st("B", 1130, 1135), st("C", 1200, null)]),
      j([st("A", null, 1200), st("B", 1230, 1235), st("C", 1300, null)])
    ]);
  });

  it("does not share bestArrivals or routeScanner", () => {
    const trips = [
      t(st("A", null, 1359), st("C", 1501, null)),
      t(st("A", null, 1400), st("B", 1430, null)),
      t(st("B", null, 1430), st("C", 1500, null))
    ];

    const raptor = RaptorAlgorithmFactory.create(trips, {}, {});
    const query = new RangeQuery(raptor, journeyFactory);
    const result = query.plan("A", "C", new Date("2018-10-16"));

    setDefaultTrip(result);

    expect(result).toEqual([
      j([st("A", null, 1359), st("C", 1501, null)]),
      j([st("A", null, 1400), st("B", 1430, null)], [st("B", null, 1430), st("C", 1500, null)]),
      j([st("A", null, 1400), st("B", 1430, null)], [st("B", null, 1430), st("C", 1500, null)]),
    ]);
  });
});
